src/client/game-over.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+
+interface GameOverProps {
+  isWinner: boolean;
+  maxCountdownTime: number;
+  // onPlayAgain: () => void;
+  onBackToMainMenu: () => void;
+}
+interface GameOverState {
+  countdownTime: number;
+}
+
+export class GameOver extends React.Component<GameOverProps, GameOverState> {
+  countdownTimer: number;
+
+  constructor(props: GameOverProps) {
+    super(props);
+
+    //not really necessary for just going back to main menu, but would be if rematch was an option
+    this.state = {countdownTime: this.props.maxCountdownTime};
+    this.countdownTimer = setTimeout(this.updateCountdown.bind(this), 1000, this.state.countdownTime - 1);
+  }
+
+  updateCountdown(time: number) {
+    this.setState({countdownTime: time});
+    if (time - 1 >= 0) {
+      this.countdownTimer = setTimeout(this.updateCountdown.bind(this), 1000, time - 1);
+    }
+    else {
+      this.goToMainMenu();
+    }
+  }
+
+  resetTimer() {
+    if (this.countdownTimer != null) {
+      clearTimeout(this.countdownTimer);
+    }
+    this.countdownTimer = null;
+  }
+
+  goToMainMenu() {
```

Comment:
What if the user wants to just go to the main menu right away? Then the timer needs to be stopped before calling the prop function